R/map_nhdplus.R

```diff
@@ -0,0 +1,193 @@
+                       lat1 = pd$plot_bbox[2,1], lat2 = pd$plot_bbox[2,2])
+
+
+  st <- get_styles(plot_config)
+
+  overlays <- NULL
+
+  if(!is.null(pd$basin)){
+
+    m  <- leaflet::addPolygons(m, data = lt(pd$basin),
+                              weight = st$basin$lwd + 2,
+                              fillColor = "transparent",
+                              color = st$basin$border,
+                              group = "basins")
+
+    overlays <- c(overlays, "basins")
+  }
+
+  if(!is.null(pd$network_wtbd)){
+
+    m  <- leaflet::addPolygons(m, data = lt(pd$network_wtbd),
+                              weight = st$network_wtbd$lwd,
+                              fillOpacity = .8,
+                              fillColor = st$network_wtbd$col,
+                              color = st$network_wtbd$border,
+                              group = "Network WB")
+
+    overlays <- c(overlays, "network WB")
+  }
+
+  if(!is.null(pd$off_network_wtbd)){
+
+    m  <- leaflet::addPolygons(m, data = lt(pd$off_network_wtbd),
+                              weight = st$off_network_wtbd$lwd,
+                              fillColor = st$off_network_wtbd$col,
+                              fillOpacity = .8,
+                              color = st$off_network_wtbd$border,
+                              group = "Off-Network WB")
+
+    overlays <- c(overlays, "off-network WB")
+
+  }
+
+  if(!is.null(pd$flowline)){
+
+    m <- leaflet::addPolylines(m, data = lt(pd$flowline),
+                         weight = st$flowline$lwd + 2,
+                         opacity = 1,
+                         color = st$flowline$border,
+                         label = ~COMID,
+                         group = "Flowlines")
+
+    overlays <- c(overlays, "flowlines")
+  }
+
+  if(!is.null(pd$outlets)) {
+
+    base <- do.call(rbind.data.frame, st$outlets)
+    base$type <- rownames(base)
+    pd$outlets$type <- ifelse(pd$outlets$type %in% base$type, pd$outlets$type,"default")
+
+    outs <- left_join(pd$outlets, base, by = "type")
+
+    pal <- leaflet::colorFactor(c(base$col), domain = base$type)
+
+    m <- leaflet::addCircleMarkers(m, data = lt(outs),
+                                   fillOpacity = 1,
+                                   fillColor    = ~col,
+                                   stroke = FALSE,
+                                   radius = ~cex*5,
+                                   group = "outlets"
+                             ) %>%
+      leaflet::addLegend(position = 'topright',
+                colors = pal(unique(outs$type)), labels = unique(outs$type),
+                title = "Outlet Type", opacity = .7
+                )
+
+    overlays <- c(overlays, "outlets")
+  }
+
+  m <-leaflet::addLayersControl(m,
+      baseGroups = c("Grayscale", "Imagery", "Terrain"),
+      overlayGroups = overlays,
+      options = leaflet::layersControlOptions(collapsed = TRUE)
+    )
+
+  print(m)
```

Comment:
plot_nhdplus() has an "actually plot" toggle, I wonder if using that approach here would make sense -- you could pass back m and play with it more in that case?

Reply:
nice - how do you like `return_map`?

Reply:
added but open to changing name